flake-modules/dev/new-plugin.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+
+import os
+import re
+from argparse import ArgumentParser
+
+# Template for default.nix
+default_nix_template = """{{ lib, ... }}:
+lib.nixvim.neovim-plugin.mkNeovimPlugin {{
+  name = "{name}";
+  originalName = "{originalName}";
+  package = "{package}";
+
+  maintainers = [ lib.maintainers.YOUR_NAME ];
+
+  settingsOptions = {{
+    exampleOption = lib.nixvim.defaultNullOpts.mkBool false ''
+      Example option for the {name} plugin.
+    '';
+  }};
+}}
+"""
+
+# Template for test file
+test_nix_template = """{{
+  empty = {{
+    plugins.{name}.enable = true;
+  }};
+}}
+"""
+
+
+def kebab_case(input_string):
+    # Convert to snake_case and remove trailing underscores
+    return (
+        re.sub(r"[\W_]+", "-", input_string)
```

Comment:
`\W` is non-word chars, right? I'm wondering if we'll end up also needing to convert camelCase -> kebab-case? If so we'll need a capture group for `((?<![A-Z])[A-Z])` or something.